File: bench/tasks/brackets/task.js
import fs from "node:fs";
function balanced(s) {
  const close = { ")": "(", "]": "[", "}": "{" };
  const stack = [];
  for (const c of s) {
    if ("([{".includes(c)) stack.push(c);
    else if (c in close && stack.pop() !== close[c]) return false;
  }
  return !stack.length;
}
for (const line of fs.readFileSync("cases.txt", "utf8").trim().split("\n"))
  console.log(balanced(line) ? "ok" : "bad");

File: bench/tasks/brackets/task.py
def balanced(s):
    close = {")": "(", "]": "[", "}": "{"}
    stack = []
    for c in s:
        if c in "([{":
            stack.append(c)
        elif c in close:
            if not stack or stack.pop() != close[c]:
                return False
    return not stack

for line in open("cases.txt"):
    print("ok" if balanced(line) else "bad")
